mono/tests/test-runner.cs

```diff
@@ -400,7 +352,7 @@ class TestInfo {
 				writer.WriteEndElement ();
 			}
 			// Now dump all failing tests
-			foreach (ProcessData pd in failed) {
+			foreach (ProcessData pd in failedortimedout) {
```

Comment:
I'd just copy this for the timedout tests and add a `Test timed out` notice to the `message` xml node below.